api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -117,6 +117,8 @@ type HostedControlPlaneSpec struct {
 
 	// Services defines metadata about how control plane services are published
 	// in the management cluster.
+	// +kubebuilder:validation:MaxItems=6
+	// +kubebuilder:validation:MinItems=4
```

Comment:
This is explained in hostedCluster.Services.
hosted control plane API is beyond the scope of this PR, this change was a side effect of some shared types to prevent hosted control plane API from breaking error cost budget